ic-cdk/src/stable.rs

```diff
@@ -0,0 +1,775 @@
+//! APIs to manage stable memory.
+//!
+//! You can check the [Internet Computer Specification](https://internetcomputer.org/docs/current/references/ic-interface-spec/#system-api-stable-memory)
+//! for a in-depth explanation of stable memory.
+// mod canister;
+// #[cfg(test)]
+// mod tests;
+
+use std::{error, fmt, io};
+
+/// WASM page size in bytes.
+pub const WASM_PAGE_SIZE_IN_BYTES: u64 = 64 * 1024; // 64KB
+
+static CANISTER_STABLE_MEMORY: CanisterStableMemory = CanisterStableMemory {};
+
+/// A trait defining the stable memory API which each canister running on the IC can make use of
+pub trait StableMemory {
+    /// Gets current size of the stable memory (in WASM pages).
+    fn stable_size(&self) -> u64;
+
+    /// Attempts to grow the stable memory by `new_pages` (added pages).
+    ///
+    /// Returns an error if it wasn't possible. Otherwise, returns the previous
+    /// size that was reserved.
+    ///
+    /// *Note*: Pages are 64KiB in WASM.
+    fn stable_grow(&self, new_pages: u64) -> Result<u64, StableMemoryError>;
```

Comment:
I want to suggest calling the parameter `additional_pages`, which is unambiguous, but the public spec also uses `new_pages`.
```suggestion
    fn stable_grow(&self, additional_pages: u64) -> Result<u64, StableMemoryError>;
```